### DISCUSSION TITLE
Can't get only patch version updates for certain dependencies to work

### DISCUSSION THREAD

**Discussion by biergit** [renovatebot/renovate#10567] [Request Help] [answered]
Hi
first thanks for the wonderful tool. I'm trying to restrict Renovate to create PRs for certain dependencies only for patch version updates but unfortunately Renovate will also create PRs for minor and major version updates:

```
...
"packageRules": [
    {
      "managers": [
        "maven"
      ]
    },
    {
      "packagePatterns": [
        "^com\\.mycompany\\.acme\\.thismodule"
      ],
      "enabled": false
    },
    {
      "excludePackagePatterns": [
        "^com\\.mycompany"
      ],
      "matchUpdateTypes": "patch",
      "registryUrls": [
        "http://publicmavenrepo"
      ]
    },
    {
      "packagePatterns": [
        "^com\\.mycompany"
      ],
      "registryUrls": [
        "http://privaterepo"
      ]
    }
  ]
...
```
Any ideas?

**Comment by viceice** [comment, marked as answer]
`matchUpdateTypes` is an array. you also need to disable other updates and reenable `patch` updates after.

**Reply by biergit** [reply]
Hi @viceice according to https://docs.renovatebot.com/renovate-schema.json `matchUpdateTypes` is either an `array` or a `string`.
I'm not sure what you mean by 
> you also need to disable other updates and reenable patch updates after

but I'll try defining the `matchUpdateTypes` as `array` first. Thank you :)

**Reply by biergit** [reply]
Ok, so changing it to an `array` did nothing

**Reply by biergit** [reply]
Ah I think I've got it. Hope this helps someone as well. The trick is as @viceice pointed out to disable `major` and `minor` and also use `"separateMinorPatch": true` :

```
...
"separateMinorPatch": true,
"packageRules": [
    {
      "managers": [
        "maven"
      ]
    },
    {
      "packagePatterns": [
        "^com\\.mycompany\\.acme\\.thismodule"
      ],
      "enabled": false
    },
    {
      "packagePatterns": [
        "^com\\.mycompany"
      ],
      "registryUrls": [
        "http://privaterepo"
      ]
    },
    {
      "matchUpdateTypes": [ "major", "minor"],
      "enabled": false
    },
    {
      "excludePackagePatterns": [
        "^com\\.mycompany"
      ],
      "matchUpdateTypes": "patch",
      "registryUrls": [
        "http://publicmavenrepo"
      ]
    },
  ]
...
